### PR TITLE
Clean loaded info for released collection/partition

### PR DESCRIPTION
**Pull request by congqixia**
See also #20898 #21299

/kind improvement

Signed-off-by: Congqi Xia <congqi.xia@zilliz.com>

### CODE REVIEW

File: internal/querycoordv2/observers/collection_observer.go [hunk 1]
```diff
@@ -128,7 +131,12 @@ func (ob *CollectionObserver) observeTimeout() {
 			}
 		}
 	}
+	// Remove released collection
+	for _, collID := range obColls.Collect() {
```
Comment (yah01) [inline internal/querycoordv2/observers/collection_observer.go:135]: `for collID := range obColls` works

File: internal/querycoordv2/observers/collection_observer.go [hunk 2]
```diff
@@ -169,6 +178,11 @@ func (ob *CollectionObserver) observeTimeout() {
 			}
 		}
 	}
+
+	// Remove released partition
+	for _, partID := range obParts.Collect() {
```
Comment (yah01) [inline internal/querycoordv2/observers/collection_observer.go:183]: Same as line 135

File: internal/querycoordv2/observers/collection_observer.go (hunk 1 above)
Comment (congqixia) [reply internal/querycoordv2/observers/collection_observer.go]: fixed

File: internal/querycoordv2/observers/collection_observer.go (hunk 2 above)
Comment (congqixia) [reply internal/querycoordv2/observers/collection_observer.go]: fixed

**Pull request merged**